content/blog/2019/secure-egress-control-istio/index.md

```diff
@@ -0,0 +1,323 @@
+[direct TLS traffic to external services through an egress gateway](/docs/examples/advanced-gateways/egress-gateway/#egress-gateway-for-https-traffic).
+(To support wildcard domains, you must create
+[a custom version of an egress gateway](/docs/examples/advanced-gateways/wildcard-egress-hosts/)). Alternatively, you
+can [direct HTTP traffic through an egress gateway](/docs/examples/advanced-gateways/egress-gateway/#egress-gateway-for-http-traffic)
+and [let the egress gateway perform TLS origination](/docs/examples/advanced-gateways/egress-gateway-tls-origination/#perform-tls-origination-with-an-egress-gateway).
+
+In all the cases you have to apply some
+[additional security mechanisms](/docs/examples/advanced-gateways/egress-gateway/#additional-security-considerations),
+like Kubernetes Network Policies or an L3 firewall that will enforce that traffic from the cluster to the outside is
+allowed for the egress gateway only. See
+[an example for Kubernetes Network Policies configuration](/docs/examples/advanced-gateways/egress-gateway/#apply-kubernetes-network-policies).
+
+You must also increase the security measures applied to Istio control plane and the
+egress gateway by running them on nodes separate from the application nodes, in a separate namespace, monitoring them
+more thoroughly, etc. After all, if the attackers are able to attack Istio Mixer or the egress gateway, they could
+violate any policy.
+
+Once you direct egress traffic through an egress gateway and apply the additional security mechanisms,
+you can securely monitor the traffic and define security policies on it.
+If the application sends HTTP requests and the egress gateway performs TLS origination, you can monitor HTTP
+information, e.g HTTP methods, headers and URL paths, and you can
+[define policies](/blog/2018/egress-monitoring-access-control) based on the HTTP information. If the application
+performs TLS origination, for TLS traffic you can
+[monitor SNI and the service account](/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/) of the
+source pod, and define policies based on them.
+
+The following diagram shows Istio Security Architecture, augmented with L3 Firewalls (part of the
+[additional security mechanisms](/docs/examples/advanced-gateways/egress-gateway/#additional-security-considerations)
+provided outside of Istio by the cluster/cloud provider).
+The L3 Firewalls can have a trivial configuration that would allow incoming traffic into Istio ingress gateway pods only and
+outgoing traffic from Istio egress gateway pods only. Note that the Istio proxy of the egress gateway performs
+policy enforcement and reporting in the same way as the sidecar proxies in the application pods.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewalls" >}}
+
+Now let's examine possible attacks and let me show you how the secure egress control in Istio prevents them.
```

Comment:
how the secure -> how secure

Reply:
Fixed by https://github.com/istio/istio.io/pull/3179/commits/bcf19266a99af2f7f39da2d0fe3696d4fefcae8e